resources/metadata/npmExecuteTests.yaml

```diff
@@ -0,0 +1,92 @@
+metadata:
+  name: npmExecuteTests
+  description: Executes end-to-end tests using npm
+  longDescription: |
+    This step executes end-to-end tests in a Docker environment using npm.
+
+    The step spins up a Docker container based on the specified `dockerImage` and executes the `installScript` and `runScript` from `package.json`.
+
+    The application URLs and credentials can be specified in `appUrls` and `credentialsId` respectively. If `wdi5` is set to `true`, the step uses `wdi5_username` and `wdi5_password` for authentication.
+
+    The tests can be restricted to run only on the productive branch by setting `onlyRunInProductiveBranch` to `true`.
+
+spec:
+  inputs:
+    params:
+      - name: installCommand
+        type: string
+        description: Command to be executed for installation. Defaults to `npm ci`.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        default: "npm ci"
+      - name: runScript
+        type: string
+        description: Script to be executed from package.json for running tests. Defaults to `npm run wdi5`.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        mandatory: true
+        default: "npm run wdi5"
+      - name: appSecrets
+        type: map[string]interface{}
+        description: |
+          The object must have the following structure:
+          - `urls`: An array of objects, each representing an application URL with associated credentials.
+            Each object must have the following properties:
+            - `url`: The URL of the application.
+            - `username`: The username for accessing the application.
+            - `password`: The password for accessing the application.
+          - `username`: The base URL username.
+          - `password`: The base URL password.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        resourceRef:
+          - type: vaultSecret
+            default: appMetadata
+            name: appMetadataVaultSecretName
+      - name: onlyRunInProductiveBranch
+        type: bool
+        default: false
+        description: Boolean to indicate whether the step should only be executed in the productive branch or not.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: productiveBranch
+        type: string
+        default: "main"
+        description: The branch used as productive branch.
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
```

Comment:
Is that really needed? I would not make a simple step aware of where it's running. This should be part of the pipeline I would say.